$OpenBSD: patch-src_socket_c,v 1.6 2020/01/18 11:24:49 kn Exp $

https://sourceforge.net/p/isync/isync/ci/7607e53d56f9470ee221cd5b644dda829f54b005/

commit 7607e53d56f9470ee221cd5b644dda829f54b005
Author: Caspar Schutijser <caspar@schutijser.com>
Date:   Sun Aug 18 10:38:48 2019 +0200

    Do not crash when using Tunnel in an IPv6-enabled build
    
    socket_connected() is also called on the tunnel pipe.
    
    amends 3ceb55310.

Index: src/socket.c
--- src/socket.c.orig
+++ src/socket.c
@@ -545,8 +545,10 @@ static void
 socket_connected( conn_t *conn )
 {
 #ifdef HAVE_IPV6
-	freeaddrinfo( conn->addrs );
-	conn->addrs = 0;
+	if (conn->addrs) {
+		freeaddrinfo( conn->addrs );
+		conn->addrs = 0;
+	}
 #endif
 	conf_notifier( &conn->notify, 0, POLLIN );
 	socket_expect_read( conn, 0 );
